finagle-http/src/main/scala/com/twitter/finagle/Http.scala

```diff
@@ -107,6 +107,7 @@ object Http extends Client[Request, Response] with HttpRichClient
     val stack: Stack[ServiceFactory[Request, Response]] =
       StackClient.newStack
         .replace(TraceInitializerFilter.role, new HttpClientTraceInitializer[Request, Response])
+        .replace(StackClient.Role.prepFactory, DelayedRelease.module)
```

Comment:
`prepConn` seems to be orthogonal:

```scala
    val prepConn = new Stack.ModuleParams[ServiceFactory[Req, Rep]] {
      override def parameters: Seq[Stack.Param[_]] = Nil
      override val role = StackClient.Role.prepConn
      override val description = "Connection preparation phase as defined by a Codec"
      def make(ps: Stack.Params, next: ServiceFactory[Req, Rep]) = {
        val Stats(stats) = ps[Stats]
        val underlying = codec.prepareConnFactory(next, ps)
        new ServiceFactoryProxy(underlying) {
          val stat = stats.stat("codec_connection_preparation_latency_ms")
          override def apply(conn: ClientConnection) = {
            val begin = Time.now
            super.apply(conn) ensure {
              stat.add((Time.now - begin).inMilliseconds)
            }
          }
        }
      }
    }
```

Let's leave this for another change, if at all